lib.rs

```diff
@@ -70,7 +71,12 @@ impl<S: serde::Serializer> SerdeSerializer<S> {
 
 macro_rules! impl_m(
     ($s:expr, $key:expr, $val:expr) => ({
-        try!($s.ser_map.serialize_entry($key, $val)
+        #[cfg(feature = "opaque-keys")]
```

Comment:
If we implement `as_ref() -> &str` https://doc.rust-lang.org/std/primitive.str.html#impl-AsRef%3Cstr%3E for opaque `Key`, we wouldn't have to have conditional compilation here and `key.as_ref()` would work in both versions.

Reply:
Good catch I didn't think of that.